File: tutodoc/src/listing-full/examples/listing-full/hello-you.pl
print "Qui êtes-vous ? ";
my $name = <STDIN>;

chomp($name);

if ($name eq "") {
    print "Ah, pas très bavard aujourd'hui !\n";

} else {
    print "Bonjour $name.\n";
    print "Épatant ! En fait, pas du tout...\n";
}

File: tutodoc/src/listing-full/examples/listing-full/hello-you.hs
main :: IO ()

main = do
    putStr "Qui êtes-vous ? "
    name <- getLine

    if name == ""
        then putStrLn "Ah, pas très bavard aujourd'hui !"

        else do
            putStrLn ("Bonjour " ++ name ++ ".")
            putStrLn "Épatant ! En fait, pas du tout..."
